src/python_testing/TC_SC_3_6.py

```diff
@@ -0,0 +1,226 @@
+#
+#    Copyright (c) 2022 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+from unicodedata import name
+from matter_testing_support import MatterBaseTest, default_matter_test_main, async_test_body
+from chip.interaction_model import Status
+import chip.clusters as Clusters
+import chip.FabricAdmin
+import logging
+from mobly import asserts
+from chip.utils import CommissioningBuildingBlocks
+from chip.clusters.Attribute import TypedAttributePath, SubscriptionTransaction
+import asyncio
+import queue
+from threading import Event
+import time
+
+
+class ResubscriptionCatcher:
+    def __init__(self, name):
+        self._name = name
+        self._got_resubscription_event = Event()
+
+    async def __call__(self, transaction: SubscriptionTransaction, terminationError, nextResubscribeIntervalMsec):
+        self._got_resubscription_event.set()
+        logging.info("Got resubscription on client %s" % self.name)
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def caught_resubscription(self) -> bool:
+        return self._got_resubscription_event.is_set()
+
+
+class AttributeChangeAccumulator:
+    def __init__(self, name):
+        self._name = name
+        self._all_data = queue.Queue()
+
+    def __call__(self, path: TypedAttributePath, transaction: SubscriptionTransaction):
+        data = transaction.GetAttribute(path)
+        value = {
+            'sub_name': self._name,
+            'endpoint': path.Path.EndpointId,
+            'attribute': path.AttributeType,
+            'value': data
+        }
+        self._all_data.put(value)
+        logging.info("Got subscription report on client %s: %s" % (self.name, value))
+
+    @property
+    def all_data(self):
+        data = []
+        while True:
+            try:
+                data.append(self._all_data.get(block=False))
+            except queue.Empty:
+                break
+        return data
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+
+class TC_SC_3_6(MatterBaseTest):
```

Comment:
Please hold off till I've had a chance to review this, just to make sure it's making all the right assumptions about the Python APIs being used.